tests/test_routing.py

```diff
@@ -71,3 +77,23 @@ def test_docs_work_using_defined_root_path(
     )
     assert docs_response.status_code == expected_status_code
     assert schema_response.status_code == expected_status_code
+
+
+@pytest.mark.parametrize(
+    "test_route,expected_status_code",
+    [("/favicon.ico", 307), ("/api/favicon.ico", 307)],
+)
+def test_favicon_reachable_with_defined_root_path(
+    test_app, monkeypatch, test_route, expected_status_code
+):
+    """
+    Test that when the API root_path is set to a non-empty string, the favicon is still reachable
```

Comment:
aren't we calling this base_path now?

Reply:
as above - here `root_path` is referring to the name of the variable in FastAPI!

indeed, in our communications we refer to it as the base path